### PR TITLE
Implement `equation_of_line` and setup CI testing

### LINKED ISSUE #1: Add some example code
The repository needs some example code so that we can demonstrate branching, merge conflicts, and other Git and GitHub features in the course.

We will start out by implementing a function for the equation of a line (given two points) using the $y = mx+c$ approach that attendees may be familiar with from school.

### LINKED ISSUE #6: Set up continuous integration testing
Set up a GitHub Actions workflow for creating a Python environment and running the tests.

### PR DESCRIPTION
**Pull request by jwallwork23**
Closes #1.
Closes #6.

This PR provides an implementation for the `equation_of_line` function using the standard $y=mx+c$ formula.

It also sets up a test suite for continuous integration testing.

### CODE REVIEW

File: .github/workflows/test_suite.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+name: Test suite
+
+# Controls when the workflow will run
+on:
+
+  # Triggers the workflow on pushes to the 'main' branch
+  push:
+    branches: ["main"]
+
+  # Triggers the workflow on pushes to pull requests
```
Comment (TomMelt) [inline .github/workflows/test_suite.yml:11]: :+1: thanks for the detailed comments. It's clear to see what each bit of the workflow `yml` does

File: .github/workflows/test_suite.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+      # Checkout the repository
+      - name: Checkout code
+        uses: actions/checkout@v4
+
+      # Install Python 3
+      - name: Install Python
+        uses: actions/setup-python@v5
+        with:
+          python-version: '3.x'
```
Comment (TomMelt) [inline .github/workflows/test_suite.yml:37]: what is the benefit of `'3.x'` vs `'3.12'` for example?

**Review by TomMelt — APPROVED** [review]
thanks for add CI and implementing `equation_of_line`. Looks good to me. I have some general questions but I am happy for the merge to go ahead.

**Comment by jwallwork23** [comment]
Thanks for the review @TomMelt.

**Pull request merged**